### PR TITLE
Feature / Claim Wallet Rewards

### PR DESCRIPTION
**Pull request by gergana95**
Create sign account op function in transfer helpers to claim wallet rewards.

### CODE REVIEW

**Review by Ivshti — CHANGES_REQUESTED** [review]
I don't think mainCtrl  should get even more cluttered than it is and gain functionalities like this - let's think of a better solution

File: src/controllers/main/main.ts [hunk 1]
```diff
@@ -681,6 +681,33 @@ export class MainController extends EventEmitter {
     await this.addUserRequest(userRequest)
   }
 
+  async claimWallet(
+    selectedToken: TokenResult,
+    claimableRewardsData: any,
```
Comment (PetromirDev) [inline src/controllers/main/main.ts:686]: replace any

File: src/libs/portfolio/interfaces.ts [hunk 2]
```diff
@@ -87,7 +87,9 @@ interface Total {
 type AdditionalPortfolioProperties = 'updateStarted' | 'tokens'
 
 // Create the final type with some properties optional
-export type AdditionalPortfolioNetworkResult = Partial<PortfolioLibGetResult> &
+export type AdditionalPortfolioNetworkResult = Partial<PortfolioLibGetResult> & {
+  claimableRewardsData: any
```
Comment (PetromirDev) [inline src/libs/portfolio/interfaces.ts:91]: just add it next to total. Also replace any

File: src/libs/transfer/userRequest.ts [hunk 3]
```diff
@@ -14,6 +16,38 @@ interface BuildUserRequestParams {
...
 }
 
+const supplyControllerAddress = '0xA69B8074CE03A33B13057B1e9D37DCDE0024Aaff'
+const WALLET_STAKING_ADDR = '0x47Cd7E91C3CBaAF266369fe8518345fc4FC12935'
+
+function buildClaimWalletRequest({
+  selectedAccount,
+  selectedToken,
+  claimableRewardsData,
+}: { selectedAccount: string, selectedToken: TokenResult, claimableRewardsData: any }): SignUserRequest | null {
```
Comment (PetromirDev) [inline src/libs/transfer/userRequest.ts:26]: replace any

File: src/libs/transfer/userRequest.ts [hunk 4]
```diff
@@ -14,6 +16,38 @@ interface BuildUserRequestParams {
   recipientAddress: string
 }
 
+const supplyControllerAddress = '0xA69B8074CE03A33B13057B1e9D37DCDE0024Aaff'
+const WALLET_STAKING_ADDR = '0x47Cd7E91C3CBaAF266369fe8518345fc4FC12935'
```
Comment (PetromirDev) [inline src/libs/transfer/userRequest.ts:20]: Can we add these constants to some common file?

**Review by PetromirDev — CHANGES_REQUESTED** [review]
Fix tests

File: src/libs/transfer/userRequest.ts [hunk 5]
```diff
@@ -1,11 +1,14 @@
 import erc20Abi from 'adex-protocol-eth/abi/ERC20.json'
 import { Interface, parseUnits } from 'ethers'
-import { SignUserRequest } from 'interfaces/userRequest'
+import { Call, SignUserRequest } from 'interfaces/userRequest'
```
Comment (PetromirDev) [inline src/libs/transfer/userRequest.ts:3]: fix import

**Review by PetromirDev — CHANGES_REQUESTED** [review]

**Pull request merged**